credential issuance: clarification about user auth at authz endpoint and idp hint (#1187)
italia/eidas-it-wallet-docs: docs/en/credential-issuance-endpoint.rst

            - **type**: it MUST be set to ``openid_credential``,
            - **credential_configuration_id**: JSON String. String specifying a unique identifier of the Credential in a specific format that MUST be mapped in the `credential_configurations_supported` metadata claim of the Credential Issuer. For instance,``dc_sd_jwt_pid`` can be used for PID in SD-JWT VC format, ``dc_sd_jwt_mDL`` for mobile driving licence in SD-JWT VC format and ``mso_mdoc_mDL`` for mobile driving license in mdoc format.

        When the Wallet Instance intends to indicate eID Substantial Authentication with MRTD Verification as an optional hint, an additional JSON Object MAY be included with the following claims:

            - **type**: REQUIRED. MUST be ``it_l2+document_proof``,

>             - **idphinting**: REQUIRED. URL of the Identity Provider to be used as an optional hint for the Authorization Server, according to :ref:`credential-issuance-endpoint:User Authentication Method Selection`,

Reviewer: I think there is a technical edge case introduced by making `it_l2+document_proof` optional.

When the `it_l2+document_proof` object in `authorization_details` is omitted by the Wallet Instance, the Authorization Server may present a discovery page to the user to choose their preferred authentication method.

If the user selects the **L2+ with MRTD authentication** method on that discovery page, the AS needs key technical parameters—specifically `challenge_redirect_uri` and `challenge_method`—to complete the subsequent multi-step flow and browser confirmation. 

Since these parameters are defined inside the `it_l2+document_proof` object, it's currently unclear how the AS can execute the L2+ flow if the Wallet Instance didn't include that object in the initial PAR request.

To fix that, we could define that Authorization Server MUST NOT offer or select the L2+ method unless the necessary `it_l2+document_proof` parameters were provided by the Wallet in the initial authorization request.

Author: I can make it

@fmarino-ipzs is it ok VCI side?

Author: @manuraf I have resolved your comment here: 9794be782c43ca91647edf858de3c1b3fe527027

@fmarino-ipzs I have also reduced the implementation burden to the AS, considering that without idphinting configured with l2plus the AS cannot include l2plus into the discovery page, making materially only L3 the sole auth method available, todate, for the issuance of the PID

The authorization server MUST first verify the identity of the User that own the credential.

User Authentication Method Selection
....................................

The Authorization Server of the Credential Issuer is responsible for authenticating the User according to its authentication policies and the applicable Trust Framework. The decision on which authentication method to apply, and whether to present a discovery page of the supported methods, MUST remain with the Authorization Server of the Credential Issuer.

For PID issuance:

  * The Authorization Server MUST authenticate the User using one of the methods it supports, including CieID with Level of Assurance High (CIE L3) and, if supported, eID Substantial Authentication with MRTD Verification as defined in :ref:`credential-issuance-l2plus:eID Substantial Authentication with MRTD Verification for PID Issuance`.
  * The Authorization Server MAY present a discovery page enabling the User to select among the authentication methods it supports and allows for the request in progress.

>   * The Wallet Instance MAY include in the Pushed Authorization Request one or more optional hints related to the User's preferred authentication method, as defined below. Such hints MUST NOT be interpreted as a binding imposition of the authentication method by the Wallet Instance.

Reviewer 2: ```suggestion
  * The Wallet Instance MAY present a discovery page and MUST therefore include in the Pushed Authorization Request one or more optional hints related to the User's preferred authentication method, as defined below. Such hints MUST NOT be interpreted as a binding imposition of the authentication method by the Wallet Instance.
```

Author: MAY and MUST therefore doesn't really work together

>   * If the Authorization Server receives a hint that it supports and accepts, it SHOULD honor it and MAY omit presenting a discovery page and/or autonomously applying only its default authentication method.

Reviewer 2: ```suggestion
  * If the Authorization Server receives a hint that it supports and accepts, it SHOULD honor it and SHOUD NOT present a discovery page and/or autonomously applying only its default authentication method.
```
If an authentication hint is provided, the Authorization Server should avoid presenting an authentication method selection page, as the User has already made that choice in the Wallet.
If no authentication hint is provided, the Authorization Server may (or should?) present an authentication method selection page.
This is to preserve the user experience.

Author: the idp hint support is optional for both the wallet solutions and the credential issuers

  * If the hint is absent, unsupported, or not accepted, the Authorization Server MUST apply its authentication policies, including any discovery page and the default authentication method.

>   * A Wallet Instance MUST NOT be required to implement an authentication-method discovery interface as an interoperability prerequisite for PID issuance. Any Wallet-side pre-selection interfaces (for example to improve User experience) remain optional and MUST be limited to providing hints to the Authorization Server.

Reviewer 2: This sentence seems to me to invalidate the CIE + PIN authentication process implemented in the IO app – is that correct? cc: @manuraf

Author: resolved in 9794be782c43ca91647edf858de3c1b3fe527027 as commented by manuel

The authentication hints recognized for PID issuance are the following:

  * the presence in the ``authorization_details`` array of an object with ``type`` set to ``it_l2+document_proof`` constitutes an optional hint by which the Wallet Instance indicates a preference for eID Substantial Authentication with MRTD Verification;
  * the ``idphinting`` claim within that object constitutes an optional hint for the URL of the LoA3 Identity Provider to be used in the context of that method;

>   * the absence of the ``it_l2+document_proof`` object does not, by itself, constitute a Wallet Instance mandate on the authentication method: in that case the Authorization Server applies its policies. In the absence of a User selection via a Credential Issuer discovery page, the PID Provider default authentication method MUST be CieID with Level of Assurance High (CIE L3).

Reviewer 2: ```suggestion
  * the absence of the ``it_l2+document_proof`` object does not, by itself, constitute a Wallet Instance mandate on the authentication method: in that case the Authorization Server applies its policies. In the absence of a User selection via a Credential Issuer discovery page, the authentication cannot be performed.
```
I suggest making this change because, if the user does not make a selection, no authentication can take place. There is no default authentication method that can be applied. The process simply stops.

Author: since the discovery page is optional for the AS implementation, it shouldapply its preferred user auth method and then making the authentication method always performed

.. note::
   This Section concerns authentication method selection during PID issuance and does not apply to the Relying Party Touchpoint Discovery Page described in :ref:`functionalities:User Experience Design`, which has a distinct purpose (selection of the method to access a service).

> For (Q)EAA issuance, the Authorization Server MUST authenticate the User by requesting presentation of a valid PID from the Wallet Instance using [`OpenID4VP`_], as described in :ref:`credential-issuance-low-level:Low-Level Issuance Flow`. In that context, discovery of SPID, CIE L3, or L2+ methods does not apply.

Reviewer 2: This is incorrect if we accept that the NPID / National EID / IT-Wallet ID is obtained as a result of an authentication process rather than by presenting the PID. Am I right?

Author: I didn't get this one

may I ask you to reelaborate ore?
